Add separate cluster for workloads (#771)
kyma-project/test-infra: docs/prow/prow-architecture.md

![Prow architecture overview](./assets/prow-architecture.svg)

## Provisioning and Secrets

> The cluster where Prow runs it's workload pods get provisioned by the [set-up-workload-cluster.sh](../../prow/set-up-workload-cluster.sh). To enable the Prow cluster to schedule jobs in the workload cluster, a secret needs to be created using [create-secrets-for-workload-cluster.sh](../../prow/create-secrets-for-workload-cluster.sh) script. The Prow main cluster gets provisioned by the [install-prow.sh](../../prow/install-prow.sh) script that the Prow administrator executes manually. The scripts reads the full configuration from the `test-infra` repository. Based on that, the administrator can recreate the clusters at any time. That is why, when you make any changes to the cluster configuration, apply them directly on the repository instead of the runtime. Only the Prow administration team needs access to the runtime. All others should contribute changes through the repository with the review process in place. Any new configuration is uploaded on the cluster by administrators.

Reviewer: ```suggestion
The [set-up-workload-cluster.sh](../../prow/set-up-workload-cluster.sh) script provisions the cluster where Prow runs its workload Pods. To enable the Prow cluster to schedule jobs in the workload cluster, you must create a Secret using [create-secrets-for-workload-cluster.sh](../../prow/create-secrets-for-workload-cluster.sh) script. The Prow main cluster gets provisioned by the [install-prow.sh](../../prow/install-prow.sh) script that the Prow administrator executes manually. The script reads the full configuration from the `test-infra` repository. Based on that, the administrator can recreate the clusters at any time. That is why, when you make any changes to the cluster configuration, apply them directly on the repository instead of the runtime. Only the Prow administration team needs access to the runtime. All others should contribute changes through the repository with the review process in place. The administrators upload any new configuration to the cluster.
```

Reviewer: I don't know why it doesn't show the changes :(
